src/dockleHelper.ts

```diff
@@ -0,0 +1,94 @@
+import * as os from 'os';
+import * as path from 'path';
+import * as util from 'util';
+import * as fs from 'fs';
+import * as toolCache from '@actions/tool-cache';
+import * as core from '@actions/core';
+
+const stableDockleVersion = "0.2.4";
+const dockleLatestReleaseUrl = "https://api.github.com/repos/goodwithtech/dockle/releases/latest";
+const dockleToolName = "dockle";
+
+export async function getDockle(): Promise<string> {
+    const latestDockleVersion = await getLatestDockleVersion();
+    let cachedToolPath = toolCache.find(dockleToolName, latestDockleVersion);
+    if (!cachedToolPath) {
+        let dockleDownloadPath;
+        const dockleDownloadUrl = getDockleDownloadUrl(latestDockleVersion);
+        const dockleDownloadDir = `${process.env['GITHUB_WORKSPACE']}/_temp/tools/dockle`;
+        console.log(util.format("Could not find dockle in cache, downloading from %s", dockleDownloadUrl));
+
+        try {
+            dockleDownloadPath = await toolCache.downloadTool(dockleDownloadUrl, dockleDownloadDir);
+        } catch (error) {
+            throw new Error(util.format("Failed to download dockle from %s", dockleDownloadUrl));
+        }
+
+        fs.chmodSync(dockleDownloadPath, "777");
+        const untarredDocklePath = await toolCache.extractTar(dockleDownloadPath);
+        cachedToolPath = await toolCache.cacheDir(untarredDocklePath, dockleToolName, latestDockleVersion);
+    }
+
+    const dockleToolPath = findDockle(cachedToolPath);

```

Comment:
Do we need this logic? Can't we simply assume that binary will be present inside the extracted folder?

Reply:
This was for the case when we find the tool in cache and don't have to download it. But I think even in that case, the cachedToolPath (something like /opt/hostedtoolcache/trivy/0.5.2/x64) is unique enough that we don't need a function to find the tool in that path. I'll remove the function and add cachedToolPath+"/"+toolName.